application_sdk/application/__init__.py

```diff
@@ -167,19 +182,46 @@ async def setup_server(
         has_configmap: bool = False,
     ):
         """
-        Optionally set up a server for the application. (No-op by default)
+        Set up FastAPI server and automatically mount MCP if enabled.
         """
         if self.workflow_client is None:
             await self.workflow_client.load()
 
-        # Overrides the application server. serves the UI, and handles the various triggers
+        # Create MCP ASGI app first if enabled (to get lifespan)
+        mcp_asgi_app: Optional[Any] = None
+        if self.enable_mcp:
+            try:
+                mcp_asgi_app = await self._create_mcp_asgi_app()
+            except (ValueError, ImportError) as e:
+                logger.error(f"Failed to create MCP server: {e}")
+                raise
+
+        # Setup FastAPI server with MCP lifespan
         self.server = APIServer(
+            lifespan=mcp_asgi_app.lifespan if mcp_asgi_app else None,
             workflow_client=self.workflow_client,
             ui_enabled=ui_enabled,
             handler=self.handler_class(client=self.client_class()),
             has_configmap=has_configmap,
         )
 
+        # Mount MCP at root (clean approach)
+        if self.enable_mcp and mcp_asgi_app:
+            try:
+                self.server.app.mount("", mcp_asgi_app)  # Mount at root
+                logger.info("Mounted MCP at root")
+                logger.info("MCP Debug Info:")
+                logger.info("   • MCP endpoint: http://localhost:8000/mcp")
+                logger.info("   • Transport: streamable_http")
+                logger.info("   • Debug with MCP Inspector using the above URL")
+            except Exception as e:
+                logger.error(f"Failed to mount MCP at root: {e}")
+                # Fallback: mount at subpath
+                self.server.app.mount("/tools", mcp_asgi_app)
+                logger.info("Fallback: Mounted MCP at /tools")
+                logger.info("   • MCP endpoint: http://localhost:8000/tools/mcp")
```

Comment:
(nit) do we need multiple logger operations? let's make each section have just a single line logger usage